library/pkparse.c

```diff
@@ -545,7 +546,18 @@ static int pk_get_rsapubkey( unsigned char **p,
 
     if( ( ret = mbedtls_rsa_import_raw( rsa, *p, len, NULL, 0, NULL, 0,
                                         NULL, 0, NULL, 0 ) ) != 0 )
-        return( MBEDTLS_ERR_PK_INVALID_PUBKEY );
+    {
+        if( ret == MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED )
+        {
+            ret = MBEDTLS_ERR_PK_INVALID_PUBKEY |
+                  MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED;
```

Comment:
Can these be or'd together like this? Do users use them as bitfields?

Reply:
The Or here is basically the high level and the low level. If I remember correct, the `MBEDTLS_ERR_PK_INVALID_PUBKEY` is still used in caller functions

Reply:
In principle, yes: an error code from a “high-level” module of Mbed TLS can be either a high-level error code, or a bitwise-or of a high-level error code and a low-level error code. See `error.h`.

In practice, I've seen bugs before from code expecting a high-level error code and getting one bitwise-ored with a low-level error code. So I wouldn't be surprised if this broke some existing code. We need to document this change clearly and in a way that people will find. This used to be reasonable code (abbreviated), but no longer is:
```
switch (mbedtls_pk_parse_key(…)) {
    case 0: do stuff; break;
    case MBEDTLS_ERR_PK_INVALID_PUBKEY: /*algorithm not supported, try the next key*/ continue;
    default: printf("Fatal error parsing key. Giving up.\n"); exit(1);
}
```